packages/imperative/src/events/src/doc/IImperativeEventJson.ts

```diff
@@ -40,3 +40,5 @@ export interface IImperativeEventJson {
      */
     user?: boolean;
 }
+
+export type ImperativeEventCallback = () => void | Promise<void>;
```

Comment:
We might want to change this to `PromiseLike` in order to support callbacks from a VS Code extension with a `Thenable` return type (as Microsoft's `Thenable` satisfies the `PromiseLike` type constraint, but does not satisfy `Promise`).

Reply:
Oh, that's right, Thenables (as PromiseLike) don't implement a `catch` method 😋